api/types/kustomization.go

```diff
@@ -126,6 +126,10 @@ type Kustomization struct {
 	// the map will have a suffix hash generated from its contents.
 	ConfigMapGenerator []ConfigMapArgs `json:"configMapGenerator,omitempty" yaml:"configMapGenerator,omitempty"`
 
+	// GkeSaGenerator is a list of arguments to generate
```

Comment:
Let's keep this out of kustomization for now, and use it via the generators field only.

The API  needs to settle first.